src/memory.hpp

```diff
@@ -453,17 +458,26 @@ struct cvk_sampler : public _cl_sampler, api_object<object_magic::sampler> {
     cl_addressing_mode addressing_mode() const { return m_addressing_mode; }
     cl_filter_mode filter_mode() const { return m_filter_mode; }
     VkSampler vulkan_sampler() const { return m_sampler; }
+    VkSampler vulkan_sampler_norm() {
+        if (m_sampler_norm == VK_NULL_HANDLE) {
+            if (!init(true)) {
+                return VK_NULL_HANDLE;
+            }
+        }
```

Comment:
I don't really like this idiom as currently written. Having what looks like a getter reinitialise the sampler object breaks the principle of least surprise. Since we can't know at sampler creation time whether we need to force enable normalized coordinates we can either:
1. Always create both samplers. It's a bit wasteful but probably not a practical problem.
2. Give this function a more explicit name. Something along the lines of `get_or_create_vulkan_sampler_with_normalized_coords` (or better name).

Reply:
I went with option 2.